### DISCUSSION TITLE
Watcher's acidemitters contact with all characters and lag

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#15102] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

This trigger from `acidemitter` in `alienitems.xml` spawned by hurt watcher do not burn monsters, but still creates a funny amount of contacts with them and lag
![Screenshot_4](https://github.com/user-attachments/assets/7c8e30c7-59e3-4a32-b557-88c4a3eb6faf)

https://github.com/user-attachments/assets/56e50554-c785-4fdf-967b-65e2300d1aff

https://github.com/user-attachments/assets/85edf23c-75c7-47a4-bbd3-e3c94c38a18b

### Reproduction steps

spawn some mosters and `acidemitter`s

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
If you're wondering if this is fixed by "Changes to the physics engine to improve collision detection performance", it's not

**Comment by Regalis11** [comment]
It seems that the contacts or the physics were not really the main issue here. The cost of the trigger checks is fairly minimal (and imo not worth optimizing), but the watchers seemed to spawn up to 9 acid emitters every time they took damage, which could lead to enormous amounts of active TriggerComponents. Addressed in our private repo now, fix coming up in a future build.
